src/libPersistence/BlockStorage.cpp

```diff
@@ -533,53 +533,74 @@ bool BlockStorage::GetDSCommittee(
   return true;
 }
 
-bool BlockStorage::PutShardStructure(const DequeOfShard& shards,
-                                     const uint32_t myshardId) {
+bool BlockStorage::PutShardStructure(const uint64_t& blockNum,
```

Comment:
For backward-compatible, it's better to create some new API/DB to store the historical sharding structure, for both write/read sides.